revolutionpi/digital_interrupts.go

```diff
@@ -75,16 +80,22 @@ func initializeDigitalInterrupt(pin SPIVariable, g *gpioChip) (*digitalInterrupt
 
 	// check if the pin is configured as a counter
 	// b[0] == 0 means the interrupt is disabled, b[0] == 3 means the pin is configured for encoder mode
-	if b[0] == 0 || b[0] == 3 {
-		return &digitalInterrupt{}, errors.New("pin is not configured as a counter")
+	if b[0] == 0 || (b[0] == 3 && !isEncoder) {
```

Comment:
Do we want to check the other way, too? If `b[0]` indicates a normal interrupt but `isEncoder` is true, is that a problem?

Reply:
so I added the `di.isEncoder = b[0] == 3` logic to handle this, so anything trying to use an encoder can validate that the pin is an encoder. Wasn't sure if this was better than an explicit error though and I can be convinced to make it an error instead

Reply:
If a user sets up an encoder in their config, but `b[0]` indicates it's not an encoder, we're not going to treat it like one, and the user will be surprised. I'd make it an explicit error. (a doubt: I prefer explicit errors for all kinds of things that other people don't, and I might be overzealous here. You know this code better than I do: if you think this is a bad idea, leave as-is.)

Reply:
nah you right, I was using a later check on isEncoder to see if it was an encoder, but the initialize function should fail if the pin isn't able to initialize correctly